### PR TITLE
model:  enable HIP backend for Vibe Voice TTS

### PR DESCRIPTION
**Pull request by IIIIIllllIIIIIlllll**
## Summary

Removes the ROCm/HIP restriction for VibeVoice TTS after validation on AMD hardware:

- `src/models/vibevoice/session.cpp`: add `BackendType::Hip` to the session
  backend whitelist; align the voice prompt duration cap with CUDA (30 s).
- `src/framework/modules/conv_modules.cpp`: enable the conv-transpose1d
  col2im fast path for HIP (used by the VibeVoice acoustic decoder).
- `docs/HIP.md`: move VibeVoice TTS out of the "not yet enabled" list and
  record the validation results.

## Validation (gfx1151 / Strix Halo iGPU, ROCm 7.14, Linux)

- Prompt-side fingerprints (text encoding, acoustic encoder, prompt
  embeddings, prefill logits + top-5) match CPU and CUDA within ~1e-3
  relative float noise.
- ggml `test-backend-ops` on ROCm0: **11953 OK / 0 FAIL**.
- CUDA graphs on/off and op fusion on/off produce bit-identical output.
- End-to-end TTS generation works and produces clean audio.

Note on numerical parity: with identical inputs (same seed + injected
diffusion noise), HIP, CUDA, and CPU all agree for the first ~34
autoregressive steps, then diverge at a near-tie argmax step. Crucially,
**CUDA diverges from CPU the same way** (one step earlier than HIP), so
this is inherent bf16 summation-order noise amplified by autoregressive
generation — not a HIP-specific defect.

AI usage : I reviewed the VibeVoice logic, designed a workflow, and had Kimi K3 run tests separately on my servers (equipped with 2080TI, 3090, and GFX1151). I confirmed that there are no significant differences between the HIP and CUDA backends.

Output:
[out_hip.wav](https://github.com/user-attachments/files/30581545/out_hip.wav)

### CODE REVIEW

File: src/models/vibevoice/session.cpp [hunk 1]
```diff
@@ -46,9 +46,10 @@ void validate_weight_storage(engine::assets::TensorStorageType storage_type, con
 const runtime::SessionOptions & require_supported_backend_options(const runtime::SessionOptions & options) {
     if (options.backend.type != engine::core::BackendType::Cpu &&
         options.backend.type != engine::core::BackendType::Cuda &&
+        options.backend.type != engine::core::BackendType::Hip &&
         options.backend.type != engine::core::BackendType::Vulkan &&
         options.backend.type != engine::core::BackendType::Metal) {
-        throw std::runtime_error("VibeVoice session supports only CPU, CUDA, Vulkan, and Metal backends");
+        throw std::runtime_error("VibeVoice session supports only CPU, CUDA, HIP, Vulkan, and Metal backends");
     }
```
Comment (francescobozzo) [inline src/models/vibevoice/session.cpp:53]: Given these are all the backends, can we just remove this check?

Comment (IIIIIllllIIIIIlllll) [reply src/models/vibevoice/session.cpp:53]: Of course, that's how the original author wrote it, and I didn't change it.

**Comment by 0xShug0** [comment]
@IIIIIllllIIIIIlllll Merged, thanks!
